pkg/api/types.go

```diff
@@ -396,6 +398,24 @@ type NFSVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the lifetime of a pod
+type GlusterfsVolumeSource struct {
+	// Required: hosts are Glusterfs hosts address array
+	Hosts []string `json:"hosts"`
+
+	// Required: Path is the Glusterfs volume path
+	Path string `json:"path"`
+
+	// Optional: mountOptions is the mount time options
+	// including readonly
+	MountOpt string `json:"mountOptions,omitempty"`
```

Comment:
*readonly* is an option, i.e. *-o rw*, although this is necessary for disk based volumes (pd or iscsi, for the sake of bind mount), singling it out as a special option for a filesystem mount, IMHO, is unnecessary.